lib/home/profile/account/close_account.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:flutter/material.dart';
+import 'package:thc/firebase/user.dart';
+import 'package:thc/utils/navigator.dart';
+
+class _DeleteSuccess extends StatelessWidget {
+  const _DeleteSuccess();
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      title: const Text('Account Closed'),
+      content: const Text('You have successfully closed your account.'),
+      actions: [
+        ElevatedButton(onPressed: () => navigator.logout(), child: const Text('OK')),
+      ],
+    );
+  }
+}
+
+class CloseAccount extends StatefulWidget {
+  const CloseAccount({super.key});
+
+  @override
+  State<CloseAccount> createState() => _CloseAccountState();
+}
+
+class _CloseAccountState extends State<CloseAccount> {
+  bool canDelete = false;
+
+  VoidCallback? get delete {
+    if (!canDelete) return null;
+
+    void disableTouch() => navigator.showDialog(
+          const SizedBox.shrink(),
+          barrierColor: Colors.transparent,
+          barrierDismissible: false,
+        );
+
+    return () async {
+      navigator.pop();
+      disableTouch();
+      await user!.yeet();
+      navigator.showDialog(const _DeleteSuccess(), barrierDismissible: false);
+    };
```

Comment:
This callback uses the newly-created `.yeet()` method, which removes all user data from Firebase.